src/util/struct_array.js

```diff
@@ -139,35 +130,18 @@ class StructArray {
         return {
             length: array.length,
             arrayBuffer: array.arrayBuffer,
-            type: {
-                members: array.constructor.prototype.members,
-                alignment: array.constructor.prototype.StructType.prototype.alignment
-            },
-            cacheKey: array.constructor.prototype._cacheKey
         };
     }
 
-    static deserialize(input: SerializedStructArray): StructArray {
-        const Klass = input.cacheKey && input.cacheKey in structArrayTypeCache ?
-            structArrayTypeCache[input.cacheKey] :
-            createStructArrayType(input.type);
-        const structArray = Object.create(Klass.prototype);
+    static deserialize(input: SerializedStructArray) {
```

Comment:
What work remains to eliminate the need for custom serialization for `StructArray`?

Reply:
Really the main issue is the need for `_refreshViews()` after deserializing.  If we added some kind of `afterDeserialize` (or maybe `initialize`) hook, I think that would be sufficient.